staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -2177,11 +2176,29 @@ func (d *awsDisk) waitForAttachmentStatus(status string) (*ec2.VolumeAttachment,
 		if attachmentStatus == "" {
 			attachmentStatus = "detached"
 		}
+		if attachment != nil && expectedDevice != "" {
+			device := aws.StringValue(attachment.Device)
+			if device != "" && device != expectedDevice {
```

Comment:
Should we also check for `instanceID` here? It is possible that device name is same as expected device but `describeVolume` returned stale attachment information from another instance?

Reply:
Added check for instanceID